### PR TITLE
handle errors in findGVR()

### PR DESCRIPTION
**Pull request by asalkeld**
Note: punted on "Convert kube api errors to CloudErrors within the kubeaction module" - needs more thought

### CODE REVIEW

File: pkg/frontend/frontend_test.go [hunk 1]
```diff
@@ -4,178 +4,14 @@ package frontend
...
 	"github.com/gorilla/mux"
 	"github.com/sirupsen/logrus"
-	"github.com/sirupsen/logrus/hooks/test"
-	kerrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	"github.com/Azure/ARO-RP/pkg/api"
 )
 
-func TestAdminReply(t *testing.T) {
```
Comment (jim-minter) [inline pkg/frontend/frontend_test.go:26]: worth salvaging any of this for a TestReply() ?

File: pkg/frontend/kubeactions/kubeactions.go [hunk 2]
```diff
@@ -47,7 +48,7 @@ func New(log *logrus.Entry, env env.Interface) Interface {
 	}
 }
 
-func (ka *kubeactions) findGVR(apiresourcelist []*metav1.APIResourceList, groupKind, optionalVersion string) []*schema.GroupVersionResource {
+func (ka *kubeactions) findGVR(apiresourcelist []*metav1.APIResourceList, groupKind, optionalVersion string) (*schema.GroupVersionResource, error) {
```
Comment (jim-minter) [inline pkg/frontend/kubeactions/kubeactions.go:51]: definitely +1 on this change

File: pkg/frontend/frontend_test.go (hunk 1 above)
Comment (asalkeld) [reply pkg/frontend/frontend_test.go:26]: can do

**Comment by asalkeld** [comment]
> Definitely onboard with the GVR change. Re the kube errors, can't decide which is better - wdyt?

While there is less code in adminReply() to deal with errors, I think it is better to contain the logic in kubeactions so it doesn't bleed everywhere. I'll try to make it prettier..

**Comment by asalkeld** [comment]
I'll separate these out to get the findGVR change in

**Pull request merged**